core/src/test/java/org/frankframework/http/authentication/OAuthAccessTokenKeycloakTest.java

```diff
@@ -50,45 +51,54 @@ public class OAuthAccessTokenKeycloakTest {
 	public static Stream<Arguments> parameters() {
 		return Stream.of(
-				Arguments.of(true, OAuthAccessTokenManager.AuthenticationType.REQUEST_PARAMETER),
-				Arguments.of(false, OAuthAccessTokenManager.AuthenticationType.REQUEST_PARAMETER),
-				Arguments.of(true, OAuthAccessTokenManager.AuthenticationType.AUTHENTICATION_HEADER),
-				Arguments.of(false, OAuthAccessTokenManager.AuthenticationType.AUTHENTICATION_HEADER)
+				Arguments.of(AbstractHttpSession.OauthAuthenticationMethod.CLIENT_CREDENTIALS_BASIC_AUTH, true),
+				Arguments.of(AbstractHttpSession.OauthAuthenticationMethod.CLIENT_CREDENTIALS_QUERY_PARAMETERS, true),
+
+				// Keycloak does not store passwords directly, so resource owner password credentials is impossible.
+				Arguments.of(AbstractHttpSession.OauthAuthenticationMethod.RESOURCE_OWNER_PASSWORD_CREDENTIALS_BASIC_AUTH, false),
+				Arguments.of(AbstractHttpSession.OauthAuthenticationMethod.RESOURCE_OWNER_PASSWORD_CREDENTIALS_QUERY_PARAMETERS, false)
 		);
 	}
 
+	@Override
+	public HttpSender createSender() throws Exception {
+		HttpSender sender = new HttpSender();
+
+		sender.setName("Http Sender");
+		sender.setTokenEndpoint(getTestEndpoint());
+		sender.setUrl("http://localhost");
+		sender.setClientId(CLIENT_ID);
+		sender.setClientSecret(CLIENT_SECRET);
+		sender.setScope("email");
+		sender.setUsername("fakeCredentialUserName");
+		sender.setPassword("fakeCredentialPassword");
+		sender.setTimeout(120000);
+
+		sender.configure();
+		sender.start();
+
+		return sender;
+	}
+
 	@MethodSource("parameters")
 	@ParameterizedTest
-	void testGetAccessToken(boolean useCredentials, OAuthAccessTokenManager.AuthenticationType authenticationType) throws Exception {
-		TestableOAuthAccessTokenManager tokenManager = getTokenManager(useCredentials, authenticationType);
-
-		TokenRequest tokenRequest = tokenManager.createRequest(credentials);
-		HTTPRequest httpRequest = tokenRequest.toHTTPRequest();
-
-		try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
-			HttpPost apacheRequest = (HttpPost) tokenManager.convertToApacheHttpRequest(httpRequest);
-
-			try (CloseableHttpResponse apacheResponse = httpClient.execute(apacheRequest)) {
-				if (!useCredentials) {
-					// expect this to go wrong because 'direct access grants' is disabled for 'iaf-test' realm
-					assertEquals(400, apacheResponse.getStatusLine().getStatusCode());
-				} else {
-					HTTPResponse response = tokenManager.convertFromApacheHttpResponse(apacheResponse);
-					TokenResponse tokenResponse = TokenResponse.parse(response);
-					AccessTokenResponse successResponse = tokenResponse.toSuccessResponse();
-
-					AccessToken accessToken = successResponse.getTokens().getAccessToken();
-					assertNotNull(accessToken);
-				}
-			}
-		}
-	}
+	void testGetAccessToken(AbstractHttpSession.OauthAuthenticationMethod oauthAuthenticationMethod, boolean shouldResolveSuccessfully) throws Exception {
+		assumeFalse(TestAssertions.isTestRunningOnGitHub());
```

Comment:
Dit statement komt te laat -- het gaat al lang fout in de static test setup, voordat individuele test-methods worden aangeroepen.